scipy/signal/signaltools.py

```diff
@@ -361,34 +366,59 @@ def fftconvolve(in1, in2, mode="full"):
     in1 = asarray(in1)
     in2 = asarray(in2)
 
+    if axes is None:
+        pass  # skip next few checks
+    elif not axes:
+        raise ValueError('if axes is specified it must be non-empty')
+    elif min(axes) < 0:
+        raise ValueError("axes cannot be negative")
+    elif max(axes) >= min(in1.ndim, in2.ndim):
+        raise ValueError("axes should be present in in1 and in2")
+
     if in1.ndim == in2.ndim == 0:  # scalar inputs
         return in1 * in2
     elif not in1.ndim == in2.ndim:
         raise ValueError("in1 and in2 should have the same dimensionality")
     elif in1.size == 0 or in2.size == 0:  # empty arrays
         return array([])
 
+    if axes is None:
+        axes = np.arange(in1.ndim)
+        other_axes = array([], dtype=np.int_)
```

Comment:
This could be done on line 370, then the `else` clause here can be unindented and everything should hopefully work (at the cost of some tiny extra cost of unnecessary `unique` and `setdiff1d` calls that are hopefully negligible, and if not, could live inside a nested conditional above)